### ISSUE TITLE
augmentation: RandomPerspective returns NaN on a size-1 spatial axis, even at distortion_scale=0

### ISSUE THREAD

**Issue by ducha-aiki** [kornia/kornia#4787]
`RandomPerspective` builds its perspective transform from the four inclusive corners `(0, 0)`, `(W - 1, 0)`, `(W - 1, H - 1)`, `(0, H - 1)`. When `H == 1` or `W == 1` two pairs of corners coincide, the homography solve is singular, and the transform matrix and the output are NaN, even at `distortion_scale=0`. This was noted in a review of #4538 and is documented as a limitation in the `RandomPerspective` docstring, but it has no issue.

## Repro

```python
import torch, kornia, kornia.augmentation as K
print(kornia.__version__, torch.__version__)
for shape in ((1, 1, 1, 8), (1, 1, 8, 1), (1, 1, 8, 8)):
    aug = K.RandomPerspective(0.0, p=1.0)
    out = aug(torch.rand(shape))
    print(shape, "nan out:", bool(out.isnan().any()), "nan matrix:", bool(aug.transform_matrix.isnan().any()))
```

## Observed (kornia 0.9.0rc1 @ 57cd68e6f, torch 2.14.0, CPU)

```
(1, 1, 1, 8) nan out: True nan matrix: True
(1, 1, 8, 1) nan out: True nan matrix: True
(1, 1, 8, 8) nan out: False nan matrix: False
```

## Expected

A finite result: at `distortion_scale=0` the identity, and otherwise a perspective map that is well defined along the non-degenerate axis (or, if kornia chooses not to support a size-1 axis, a `ValueError` naming the input size). Silent NaN propagation is the one outcome that should not happen.

## Why this is a bug

Other geometric augmentations accept a size-1 axis, and a NaN image produced silently at zero distortion poisons a training batch without any error.

Posted on behalf of @ducha-aiki by Claude (Opus 5.5).
